Shared/LayerCacheManager.cpp

```diff
@@ -291,6 +292,15 @@ void LayerCacheManager::layerToJson(Layer* layer)
   if (markupLayer)
     layerPath = markupLayer->path();
 
+  // Get KmlLayers
+  auto kmlLayer = dynamic_cast<KmlLayer*>(layer);
+  if (kmlLayer)
+  {
+    const QUrl kmlUrl = kmlLayer->dataset()->url();
+    const QString localKmlFile = kmlUrl.toLocalFile();
+    layerPath = localKmlFile.isEmpty() ? kmlUrl.toString() : localKmlFile;
```

Comment:
There's a QUrl [isLocalFile](https://doc-snapshots.qt.io/qt5-5.11/qurl.html#isLocalFile) that would work better here. What do you think?

Something like this
```cpp
const QUrl kmlUrl = kmlLayer->dataset()->url();
layerPath = kmlUrl.isLocalFile() ? kmlUrl.toLocalFile() : kmlUrl.toString();
```